src/expr/src/linear.rs

```diff
@@ -1621,9 +1619,7 @@ pub mod plan {
                     // Error if MLT is referenced in an unsupported position.
                     if expr2.contains_temporal()
                         || *expr1
-                            != MirScalarExpr::CallUnmaterializable(
-                                UnmaterializableFunc::MzLogicalTimestamp,
-                            )
+                            != MirScalarExpr::CallUnmaterializable(UnmaterializableFunc::MzNow)
                     {
                         return Err(format!(
                             "Unsupported temporal predicate. Note: `mz_logical_timestamp()` must be directly compared to a numeric non-temporal expression; if it is compared to a non-numeric type, consider casting that type to numeric. Expression found: {}",
```

Comment:
Probably worth updating this error? I haven't read far enough to know if we are removing `mz_logical_timestamp()`, but it would be confusing to see in response to a mis-use of `mz_now`().

Reply:
Updated.